internal/controller/etcd/reconcile_status.go

```diff
@@ -27,11 +27,20 @@ func (r *Reconciler) reconcileStatus(ctx component.OperatorContext, etcdObjectKe
 	}
 	sLog := r.logger.WithValues("etcd", etcdObjectKey, "operation", "reconcileStatus").WithValues("runID", ctx.RunID)
 	originalEtcd := etcd.DeepCopy()
-	mutateETCDStatusStepFns := []mutateEtcdStatusFn{
-		r.mutateETCDStatusWithMemberStatusAndConditions,
-		r.inspectStatefulSetAndMutateETCDStatus,
-		r.setSelector,
+
+	var mutateETCDStatusStepFns []mutateEtcdStatusFn
+	if druidv1alpha1.IsEtcdRuntimeComponentCreationEnabled(etcd.ObjectMeta) {
+		mutateETCDStatusStepFns = []mutateEtcdStatusFn{
+			r.mutateETCDStatusWithMemberStatusAndConditions,
+			r.inspectStatefulSetAndMutateETCDStatus,
+		}
+	} else {
+		sLog.Info("Skipping status checks since etcd runtime component creation is disabled")
 	}
+	mutateETCDStatusStepFns = append(mutateETCDStatusStepFns,
```

Comment:
why is this always required? Do we have a VPA used in autonomous use case and will this be able to scale a static pod?

Reply:
We don't know. Maybe it could be used. Since druid doesn't know how exactly the pod template is realised as actual running pods, we cannot make any assumptions about the selector. And if a VPA is added for the Etcd resource without `status.selector` set, it throws an error and cannot scale the etcd pod resources.

Reply:
I still somehow feel that pod selector should be part of the status, for any possible usage of `/scale` subresource, unless @unmarshall you feel otherwise.

Reply:
Lets keep the reconciliation steps to what is required today. As and when more resources/configuration is required then we incrementally add/enable them.

Reply:
Sure, makes sense.

I have made the fix in https://github.com/gardener/etcd-druid/pull/1117/commits/b3f0e481946fe40b96c7d54e79f2776d23cb9578. PTAL